### DISCUSSION TITLE
How to use Enum members

### DISCUSSION THREAD

**Discussion by tscottjendev** [microsoft/alguidelines#172] [BC Best Practices]
I have been looking at the some base code and wonder why they declare a variable of type Enum when they only assign a value to the variable not change it, just to pass to the value to a function or test it in a condition.  If the value might change, sure a variable is the only way to go but why declare a variable just to use an Enum member.  
 
So what do you like better and why?  All return the same value.

```al
    procedure FirstProcedure()
    var
        PriceType: Enum "Price Type";
    begin
        //A - Assign value to variable and pass variable?
        PriceType := PriceType::Purchase;
        SomeProcedure('A', PriceType);

        //B - Use member of variable
        SomeProcedure('B', PriceType::Purchase);

        //C - Use member of Enum directly
        SomeProcedure('C', "Price Type"::Purchase);
    end;

    local procedure SomeProcedure(Method: Text; PriceType: Enum "Price Type")
    begin
        Message('%1: %2', Method, PriceType)
    end;

```

**Comment by tscottjendev** [comment]
One more thing to consider, if a field is of type Enum, should we be using 

`varX := tableEnumField::Member;`

or 

`varX := EnumName::Member;`

**Reply by waldo1001** [reply]
In my opinion: declaring a variable merely to get to the enum-member, is pointless.

simply enum::enumname::member would do the trick to get to any member without any declaration 🤷‍♂️

**Reply by mjmatthiesen** [reply]
The only reason I can see why someone would do it, is to reduce overall line length, but I don't think that's worth another variable to potentially manage.
